packages/components/src/DateTimePickers/views/DateTimeView/DateTimeView.component.js

```diff
@@ -6,49 +6,93 @@ import IconButton from '../../IconButton';
 import HeaderTitle from '../HeaderTitle';
 import theme from './DateTimeView.scss';
 
-function DateTimeView(props) {
-	const header = {
-		leftElement: <IconButton
-			icon={{
-				name: 'talend-chevron-left',
-			}}
-			aria-label="Display previous calendar month"
-		/>,
-		middleElement: <HeaderTitle
-			label="Septembre 2017"
-			button={{
-				'aria-label': 'Switch to month and year pickers view',
-			}}
-		/>,
-		rightElement: <IconButton
-			icon={{
-				name: 'talend-chevron-left',
-				transform: 'rotate-180',
-			}}
-			aria-label="Display next calendar month"
-		/>,
-	};
-
-	const bodyElement = (
-		<div className={theme.body}>
-			<div className={theme.date}>
-				<DatePicker />
-			</div>
-			<div className={theme.time}>
+/**
+ * Get the positive euclidean modulo number from a dividend and a divisor
+ * @param {number} a Dividend
+ * @param {number} b Divisor
+ * @return The positive euclidean modulo
+ */
+export function euclideanModulo(a, b) {
+	const m = ((a % b) + b) % b;
+	return m < 0 ? m + Math.abs(b) : m;
+}
+
+class DateTimeView extends React.Component {
+
+	constructor(props) {
+		super(props);
+
+		this.incrementMonthIndexDown = this.incrementMonthIndex.bind(this, -1);
+		this.incrementMonthIndexUp = this.incrementMonthIndex.bind(this, 1);
+	}
+
+	incrementMonthIndex(monthIncrement) {
+		const monthIndexIncremented = this.props.selectedMonthIndex + monthIncrement;
+		const newMonthIndex = euclideanModulo(monthIndexIncremented, 12);
+		const yearIncrement = Math.floor(monthIndexIncremented / 12);
+		const newYear = this.props.selectedYear + yearIncrement;
 
+
+		if (this.props.onSelectMonthYear !== undefined) {
```

Comment:
`if (this.props.onSelectMonthYear) {`
But you can even drop the `if`, and consider you always have it. You set it as required in PropTypes. Plus, you only use this sub component in the DatePicker, it's an internal code

Reply:
i thought it was a good rule of thumb to always have an event handler optional. Because giving (data) must not force to receive.
Yes we know here how it's used (for now), but assuming how a component is using, even if it's local, breaks the component principle of black box and with this bring coupling.
I know it's strong to say this in this case, but i don't want to transgress principles in some cases and not in others to avoid to forget applying them when needed.

Reply:
✔